src/main/java/edu/oregonstate/cs361/battleship/BattleshipModel.java

```diff
@@ -61,5 +61,10 @@
         this.computer_cruiser = new Ship("Computer_Cruiser", 3);
         this.computer_destroyer = new Ship("Computer_Destroyer", 2);
         this.computer_submarine = new Ship("Computer_Submarine", 2);
+
+        this.playerHits = new int[0];
+        this.playerMisses = new int[0];
```

Comment:
Nice catch! I'll make sure that whenever I update these arrays, I allocate memory accordingly...